### PR TITLE
Refactor Parser into more dedicated classes

### LINKED ISSUE #95: Ensure items in food/exercise bank are unique by name
So that the associated calories can be uniquely identified.

### PR DESCRIPTION
**Pull request by tlyi**
Fixes #95

### CODE REVIEW

**Review by tryyang2001 — APPROVED** [review]
Nice inheritance technique implemented!
Looks good to merge :)

File: src/main/java/seedu/duke/parser/Parser.java [hunk 1]
```diff
@@ -1,876 +1,10 @@
...
-    private String correctCommandFormatSuggestions(String... suggestions) {
-        String formattedSuggestions = MESSAGE_ERROR_INVALID_FORMAT + LS;
-        int i = 1;
-        for (String suggestion : suggestions) {
-            formattedSuggestions += i + ". " + suggestion + LS;
-            i += 1;
-        }
-        return formattedSuggestions.stripTrailing();
-    }
+public interface Parser {
```
Comment (weidak) [inline src/main/java/seedu/duke/parser/Parser.java:8]: If I'm not wrong, interfaces should end with "-able" but I dont think Parsable is a word so shd be fine right HAHAH

**Review by weidak — COMMENTED** [review]
After looking through I'm not too sure why the profile is not updating correctly as you mentioned, maybe after merging my PR can debug better? Overall good job on the refactoring and oop-ing the parser! Looking good 👍

**Comment by weidak** [comment]
> btw i see u have unresolved merge conflict which i cannot have a look because github says it is too complex :O

I think it came up after merging my PR. After merging, I noticed a problem with my storage too... Give me a while to take a look paiseh guys

**Review by weidak — APPROVED** [review]
Looks great

**Pull request merged**